napari/components/_tests/test_layer_slicer.py

```diff
@@ -0,0 +1,244 @@
+from dataclasses import dataclass
+from threading import RLock, current_thread, main_thread
+
+import pytest
+
+from napari.components import Dims
+from napari.components._layer_slicer import _LayerSlicer
+
+# cases to consider
+# - single + multiple layers that supports async (all of layers do support async)
+# - single + multiple layers that don't support async (all of layers do not support async)
+# - mix of layers that do and don't support async
+#
+# behaviors we want to test
+# scheduling logic of the slicer (and not correctness of the slice response value)
+#
+# - for layers that support async, the slice task should not be run on the main thread (we don't want to block the calling)
+# - for layers that do not support async, slicing should always be done once the method returns
+# - slice requests should be run on the main thread
+# - pending tasks are cancelled (at least when the new task will slice all layers for the pending task)
+#
+# run all tests with:
+# pytest napari/components/_tests/test_layer_slicer.py -svv
```

Comment:
Optional: might want to convert this to a docstring that could be merged with the suggested one about the fakes.